packages/cubejs-api-gateway/src/jwk.ts

```diff
@@ -51,7 +52,7 @@ export type JWKsFetcherOptions = Pick<BackgroundMemoizeOptions<any, any>, 'onBac
 
 export const createJWKsFetcher = (jwtOptions: JWTOptions, options: JWKsFetcherOptions) => {
   const fetchJwkUrl = asyncMemoizeBackground(async (url: string) => {
-    const response = await asyncRetry(() => fetch(url), {
+    const response = await asyncRetry(() => fetch(url, { agent: new ProxyAgent() }), {
```

Comment:
We need to reuse `getHttpAgentForProxySettings()` from backend-shared here instead.